spdx/v2/common/identifier.go

```diff
@@ -14,6 +14,44 @@ const (
 	documentRefPrefix = "DocumentRef-"
 )
 
+// DocumentID represents the identifier string portion of an SPDX document
+// identifier. DocumentID should be used for reference to a SPDX document.
+// DocumentIDs should NOT contain the mandatory 'DocumentRef-' portion.
+type DocumentID string
+
+// MarshalJSON returns an DocumentRef- prefixed JSON string
+func (d DocumentID) MarshalJSON() ([]byte, error) {
+	return marshal.JSON(prefixDocumentId(d))
+}
+
+// UnmarshalJSON validates DocumentRef- prefixes and removes them when processing DocumentIDs
+func (d *DocumentID) UnmarshalJSON(data []byte) error {
+	// SPDX identifier will simply be a string
+	idStr := string(data)
+	idStr = strings.Trim(idStr, "\"")
+
+	e, err := trimDocumentIdPrefix(idStr)
+	if err != nil {
+		return err
+	}
+	*d = e
+	return nil
+}
+
+// prefixDocumentId adds the DocumentRef- prefix to an document ID if it does not have one
+func prefixDocumentId(id DocumentID) string {
+	val := string(id)
+	if !strings.HasPrefix(val, spdxRefPrefix) {
+		return documentRefPrefix + val
+	}
+	return val
+}
+
+// trimDocumentIdPrefix removes the DocumentRef- prefix from an document ID string
+func trimDocumentIdPrefix(id string) (DocumentID, error) {
```

Comment:
I think this function doesn't even need to return an error now, sorry I should have realized this the first time I asked for changes 🤦